### PR TITLE
Remove useAutoItem and TL_AUTO_ITEM usage

### PR DESCRIPTION
**Pull request by fritzmg**
Removes the option to disable the `auto_item` and also removes the `TL_AUTO_ITEM` usage.

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]

File: calendar-bundle/src/Resources/contao/modules/ModuleEventlist.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ public function generate()
 		}
 
 		// Show the event reader if an item has been selected
-		if ($this->cal_readerModule > 0  && (isset($_GET['events']) || (Config::get('useAutoItem') && isset($_GET['auto_item']))))
+		if ($this->cal_readerModule > 0  && null !== Input::get('auto_item', false, true))
```
Comment (leofeyer) [inline calendar-bundle/src/Resources/contao/modules/ModuleEventlist.php]:
```suggestion
		if ($this->cal_readerModule > 0 && isset($_GET['auto_item']))
```

File: core-bundle/src/Resources/contao/controllers/FrontendIndex.php [hunk 2]
```diff
@@ -63,30 +63,9 @@ public function renderPage(PageModel $pageModel)
 		}
 
 		// Trigger the 404 page if an item is required but not given (see #8361)
-		if ($objPage->requireItem)
+		if ($objPage->requireItem && null === Input::get('auto_item', false, true))
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/controllers/FrontendIndex.php]:
```suggestion
		if ($objPage->requireItem && !isset($_GET['auto_item']))
```

File: faq-bundle/src/Resources/contao/modules/ModuleFaqList.php [hunk 3]
```diff
@@ -62,7 +62,7 @@ public function generate()
 		}
 
 		// Show the FAQ reader if an item has been selected
-		if ($this->faq_readerModule > 0 && (isset($_GET['items']) || (Config::get('useAutoItem') && isset($_GET['auto_item']))))
+		if ($this->faq_readerModule > 0 && null !== Input::get('auto_item', false, true))
```
Comment (leofeyer) [inline faq-bundle/src/Resources/contao/modules/ModuleFaqList.php]:
```suggestion
		if ($this->faq_readerModule > 0 && isset($_GET['auto_item']))
```

File: news-bundle/src/Resources/contao/modules/ModuleNewsArchive.php [hunk 4]
```diff
@@ -61,7 +61,7 @@ public function generate()
 		}
 
 		// Show the news reader if an item has been selected
-		if ($this->news_readerModule > 0 && (isset($_GET['items']) || (Config::get('useAutoItem') && isset($_GET['auto_item']))))
+		if ($this->news_readerModule > 0 && null !== Input::get('auto_item', false, true))
```
Comment (leofeyer) [inline news-bundle/src/Resources/contao/modules/ModuleNewsArchive.php]:
```suggestion
		if ($this->news_readerModule > 0 && isset($_GET['auto_item']))
```

File: news-bundle/src/Resources/contao/modules/ModuleNewsList.php [hunk 5]
```diff
@@ -60,7 +60,7 @@ public function generate()
 		}
 
 		// Show the news reader if an item has been selected
-		if ($this->news_readerModule > 0 && (isset($_GET['items']) || (Config::get('useAutoItem') && isset($_GET['auto_item']))))
+		if ($this->news_readerModule > 0 && null !== Input::get('auto_item', false, true))
```
Comment (leofeyer) [inline news-bundle/src/Resources/contao/modules/ModuleNewsList.php]:
```suggestion
		if ($this->news_readerModule > 0 && isset($_GET['auto_item']))
```

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/controllers/FrontendIndex.php [hunk 6]
```diff
@@ -61,30 +61,9 @@ public function renderPage(PageModel $pageModel)
 		}
 
 		// Trigger the 404 page if an item is required but not given (see #8361)
-		if ($objPage->requireItem)
+		if ($objPage->requireItem && !isset($_GET['auto_item']))
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/controllers/FrontendIndex.php:65]: I would revert these changes, they become obsolete (and will cause a merge conflict) with https://github.com/contao/contao/pull/4304

**Review by aschempp — DISMISSED** [review]

**Review by ausi — DISMISSED** [review]
We could make `Config::get('useAutoItem')` to always return `true`. This might make it easier for extension developers to be compatible with 4.13 and 5.0.

**Comment by fritzmg** [comment]
You could also use `Config::get('useAutoItem') ?? true` in your extension.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**